### ISSUE TITLE
Xcode 15 Beta 8: New Non-Sendable Warning for Actor-Based Realm Init

### ISSUE THREAD

**Issue by bdkjones** [realm/realm-swift#8356]
### How frequently does the bug occur?

Always

### Description

Xcode 15 Beta 8 generates a new warning when you use Realm's suggested initialization pattern for an Actor-based realm:

```
actor MyActor
{
    private var realm: Realm!   // implicitly unwrapped so we can pass `self` to -Realm(actor:) within init.

    init() {
        realm = try await Realm(actor: self)   // WARNING: Non-sendable type 'Realm' returned by implicitly asynchronous call to nonisolated function cannot cross actor boundary
    }
}
```

### Stacktrace & log output

```shell
N/A
```

### Can you reproduce the bug?

-- select --

### Reproduction Steps

Create an actor-isolated Realm by following the pattern recommended in the documentation, as above.

### Version

10.42.0

### What Atlas Services are you using?

Local Database only

### Are you using encryption?

No

### Platform OS and version(s)

macOS 13.5.2

### Build environment

Xcode version: Xcode 15 Beta 8
Dependency manager and version: SPM

**Comment by bdkjones** [comment]
Note: changing the import statement to `@preconcurrency import RealmSwift` does remove this warning, but it's not clear that this is correct—RealmSwift does offer concurrency support via actors, etc.

**Comment by aehlke** [comment]
don't you need to specify actor: in the init?

**Comment by yanniks** [comment]
This unfortunately breaks Swift 6 Language Mode in Xcode 16 Beta 1.

**Comment by nirinchev** [comment]
We're working on enabling building with Xcode in https://github.com/realm/realm-swift/pull/8618.

**Comment by tgoyne** [comment]
Updating `Realm.init` to use `#isolation` *should* fix this, but currently it just makes the compiler crash.

**Comment by qihuijia** [comment]
@tgoyne no, actually for me, it does not work. XCode16.1 iOS18.0 Realm20.0.1

**Comment by aolan** [comment]
![Image](https://github.com/user-attachments/assets/8c12265f-92b5-4630-91e7-be3d3029763a)

**Comment by bdkjones** [comment]
@aolan Realm is dead. You need to move to something else and never touch MongoDB again.
